src/Form/matching.js

```diff
@@ -940,7 +942,7 @@ const getRelatedText = (el, form, cssSelector) => {
     const label = scope.querySelector('label')
     if (label) {
         // Try searching for a label first
-        trimmedText = getTextShallow(label)
+        trimmedText = extractElementStrings(label).join(' ')
```

Comment:
If we've identified the label, it's better to actually read its contents, rather than stopping at the shallow text. This was causing a failure.